### PR TITLE
Support units_of()

### LINKED ISSUE #106: Support simple protocol imports and units_of()
Test by creating a Python version of https://chaste.cs.ox.ac.uk/trac/browser/chaste/projects/FunctionalCuration/test/TestS1S2Protocol.hpp

- [x] Test `units_of` first with `test_clamping.py:test_clamping_with_variable_units` - can be separate PR
- [x] Then LR91 model tests basic import functionality
    - [x] Top-level import of a protocol, merging model interface
    - [x] Overriding stimulus current definition - should be done by #97 but check!
    - [x] The `units_of` construct in a larger equation: `0 :: units_of(oxmeta:membrane_stimulus_current_amplitude)`
- [x] Noble model tests with time units conversion - shouldn't really need anything further implemented after the above, just check it works!
- [x] I'm not sure how much of the manifest is currently implemented in the Python FC code. Check, and open a separate issue for that if necessary - don't consider it part of this one. --> See #148 

Blocked by #95. Part of #23.

### PR DESCRIPTION
**Pull request by MichaelClerx**
Partially addresses #106

### CODE REVIEW

File: test/test_nested_protocols.py [hunk 1]
```diff
@@ -1,8 +1,5 @@
 """Test that various test protocols are executed correctly, using a TestOdeModel."""
```
Comment (jonc125) [inline test/test_nested_protocols.py:1]: I think most of the `TestOdeModel` uses have now left this file!

File: test/test_environment.py [hunk 2]
```diff
@@ -1,122 +1,157 @@
...
-        env.define_name("one", one)
-        self.assertEqual(env.look_up("one"), one)
-        two = V.Simple(2)
-        self.assertRaises(ProtocolError, env.define_name, "one", two)  # already defined
-        self.assertRaises(ProtocolError, env.overwrite_definition, "one", two)  # don't have permission to overwrite
-        self.assertRaises(ProtocolError, env.remove, "one")  # don't have permission to overwrite
-        env.allow_overwrite = True
+def test_sim_env_txt():
+    proto_file = 'test/protocols/test_sim_environments.txt'
+    proto = fc.Protocol(proto_file)
```
Comment (MichaelClerx) [inline test/test_environment.py:14]: here!

File: test/test_environment.py [hunk 3]
```diff
@@ -1,122 +1,157 @@
...
+    proto.set_output_folder('TestVariousProtocols_TestSimEnv')
+    proto.set_model(TestOdeModel(1))
```
Comment (MichaelClerx) [inline test/test_environment.py:16]: and here

File: test/test_environment.py [hunk 4]
```diff
@@ -1,122 +1,157 @@
...
 import numpy as np
-import unittest
+import pytest
 
-import fc.language.values as V
+import fc
 import fc.environment as Env
+import fc.language.values as V
 from fc.error_handling import ProtocolError
+from fc.simulations.model import TestOdeModel
```
[the remark this answers is not in the document]
Comment (jonc125) [reply test/test_environment.py:9]: Ah, OK!

File: test/test_nested_protocols.py (hunk 1 above)
Comment (MichaelClerx) [reply test/test_nested_protocols.py]: Updated it, thanks!

**Comment by MichaelClerx** [comment]
Shall I merge this then before I continue with the s1s2 part? @jonc125 ?

**Comment by jonc125** [comment]
Makes sense.

**Pull request merged**